### PR TITLE
Adding Upload loader logic to clarify intent and implementation

### PR DESCRIPTION
**Pull request by karelm**
## Checklist

-   [ ] My code follows the code style and guidelines of this project. 
-   [ ] I have added tests to cover my changes.
-   [ ] All new and existing tests passed.
-   [ ] I have tested different user stories. 
-   [ ] I have identified similar or related features and tested they are still working.
-   [ ] I have performed a self-review of my own code.
-   [ ] I have tested my changes/additions in the latest Chrome, Firefox, and Safari.
-   [ ] I have commented my code, particularly in hard-to-understand areas.

#### Staging Deployment

To visit the URL of the staging deployment, please click "Show all checks" at the bottom of this PR and click "details" next to `bufferbotbrains/cicd-buffer-publish-legacy`

### CODE REVIEW

File: packages/shared-components/Carousel/story.jsx [hunk 1]
```diff
@@ -48,15 +47,41 @@ const UpgradeModalDecorator = storyFn => (
...
-      editMode={false}
-    />
-  ))
-  .add('should carousel with slots', () => (
-    <Carousel
-      editMode
-    />
-  ));
-*/
+  .add('should show carousel with stories', () => {
```
Comment (albennett) [inline packages/shared-components/Carousel/story.jsx]: 💯 for adding a story for this!

File: packages/shared-components/Carousel/style.js [hunk 2]
```diff
@@ -31,9 +31,9 @@ export const CarouselCard = styled.div`
...
   justify-content: center;
-  margin: ${props => (props.editMode ? `${cardMargin}px` : `16px ${cardMargin}px`)};
-  height: ${$props => $props.cardHeight}px;
-  width: ${$props => $props.cardWidth}px;
+  margin: ${props => (props.largeCards ? `${cardMargin}px` : `16px ${cardMargin}px`)};
+  height: ${props => props.cardHeight ? props.cardHeight : 198}px;
+  width: ${props => props.cardWidth ? props.cardWidth : 110}px;
   background-color: ${grayLighter};
   background-size: cover;
   background-position: center;
```
Comment (karelm) [inline packages/shared-components/Carousel/style.js]: the next line after this one refers to `props.card.thumbnail_url`... which I think shouldn't really be the case, instead it should just be `props.backgroundUrl` or something similar, and be passed in `backgroundUrl={props.card.thumbnail_url}` instead, but its okay for the time being... but we shouldnt need to rely on "card" being specified at all, just a direct property of the component

File: packages/shared-components/Carousel/style.js [hunk 3]
```diff
@@ -31,9 +31,9 @@ export const CarouselCard = styled.div`
   display: flex;
   align-items: center;
   justify-content: center;
-  margin: ${props => (props.editMode ? `${cardMargin}px` : `16px ${cardMargin}px`)};
-  height: ${$props => $props.cardHeight}px;
-  width: ${$props => $props.cardWidth}px;
+  margin: ${props => (props.largeCards ? `${cardMargin}px` : `16px ${cardMargin}px`)};
```
Comment (albennett) [inline packages/shared-components/Carousel/style.js]: It didn't occur to me until recently to use props in styled components. Love it!

File: packages/story-group-composer/components/CarouselCards/index.jsx [hunk 4]
```diff
@@ -0,0 +1,288 @@
...
+      opacity: 0.8;
+    `;
+
+const PlayIcon = () => (
+  <IconWrapper>
+    <CirclePlayIcon color={grayLighter} size={{ width: '40px' }} />
+  </IconWrapper>
+);
+
+const sortCards = cards => (
```
Comment (albennett) [inline packages/story-group-composer/components/CarouselCards/index.jsx]: is this logic specific to story-group or do we need to do this in queue, too?

File: packages/story-group-composer/components/CarouselCards/index.jsx [hunk 5]
```diff
@@ -0,0 +1,288 @@
...
+
+    const {
+      currentCards: cards,
+    } = this.state;
+    const { cardWidth, cardHeight } = getCardSizes(largeCards);
+    const cardsToRender = editMode ? getCardsToShow({ cards, totalCardsToShow }) : sortCards(cards);
+    const uploadFormatsConfig = new Map(FileUploadFormatsConfigs.MEDIA); // Clone config
+
+    return cardsToRender.map(card => (
+      <CarouselCard
```
Comment (albennett) [inline packages/story-group-composer/components/CarouselCards/index.jsx]: 💯

File: packages/story-group-composer/components/CarouselCards/index.jsx [hunk 6]
```diff
@@ -0,0 +1,288 @@
...
+  removeNotifications: PropTypes.func,
+  createNewFile: PropTypes.func,
+  createImageThumbnail: PropTypes.func,
+  updateUploadProgress: PropTypes.func,
+  uploadImageComplete: PropTypes.func,
+  videoProcessingStarted: PropTypes.func,
+  videoProcessingComplete: PropTypes.func,
+};
+
+CarouselCards.defaultProps = {
```
Comment (albennett) [inline packages/story-group-composer/components/CarouselCards/index.jsx]: Should this logic belong in defaultProps? 🤔

Comment (karelm) [reply packages/story-group-composer/components/CarouselCards/index.jsx]: I'm just using it as a stub for the moment, will replace later with functions in reducers/middleware :)

**Pull request merged**